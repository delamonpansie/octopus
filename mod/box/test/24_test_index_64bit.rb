#!/usr/bin/ruby
# encoding: ASCII

$: << File.dirname($0) + '/lib'
require 'run_env'

class Env < RunEnv
  def config
    super + <<EOD
object_space[0].index[0].key_field[0].type = "NUM64"
EOD
  end

  task :setup => "box_init.lua"
  file "box_init.lua" do
    File.write "box_init.lua", <<-EOD
        local box = require'box'
        user_proc.select = box.wrap(function (shard, n)
                return 0, {shard:object_space(n):index(0):find(0ULL),
			   shard:object_space(n):index(0):find(0xffffffffffffffffULL)}
        end)
      EOD
  end
end

Env.connect_eval do
  insert ["\x00\x00\x00\x00\x00\x00\x00\x00", "foo"]
  insert ["\xff\xff\xff\xff\xff\xff\xff\xff", "bar"]
  lua 'user_proc.select', '0'
end
