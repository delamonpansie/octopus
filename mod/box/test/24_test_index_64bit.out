# box.insert(["\x00\x00\x00\x00\x00\x00\x00\x00", "foo"])
1

# box.insert(["\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF", "bar"])
1

# box.lua("user_proc.select", "0")
[["\x00\x00\x00\x00\x00\x00\x00\x00", "foo"], ["\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF", "bar"]]

